fe1-web/src/features/wallet/network/__tests__/DigitalCashMessageApi.test.tsx

```diff
@@ -0,0 +1,132 @@
+import { Transaction } from 'features/wallet/objects/transaction';
+import { DigitalCashStore } from 'features/wallet/store';
+
+import { requestSendTransaction, requestCoinbaseTransaction } from '../DigitalCashMessageApi';
+
+jest.mock('features/wallet/store/DigitalCashStore');
+const getTransactionsByPublicKeyMock = DigitalCashStore.getTransactionsByPublicKey as jest.Mock;
+
+jest.mock('core/network/JsonRpcApi');
+const publishMock = publish as jest.Mock;
+
+const mockPopTokenKeyPair = PopToken.fromState(mockKeyPair.toState());
+
+const mockPostCoinbase = new PostTransaction({
+  transaction_id: new Hash(mockCBHash),
+  transaction: mockCoinbaseTransactionJSON,
+});
+const mockCoinbaseState = Transaction.fromJSON(mockCoinbaseTransactionJSON, mockCBHash).toState();
+
+const mockTransaction = Transaction.fromState(mockTransactionState);
+const mockPost = new PostTransaction({
+  transaction_id: mockTransaction.transactionId,
+  transaction: mockTransaction.toJSON(),
+});
+
+const checkMessageCorrectness = (message: MessageData) => {
+  expect(message.object).toBe(ObjectType.COIN);
+  expect(message.action).toBe(ActionType.POST_TRANSACTION);
+};
+
+beforeAll(() => {
+  configureTestFeatures();
+});
+
+beforeEach(() => {
+  publishMock.mockClear();
+  getTransactionsByPublicKeyMock.mockClear();
+  getTransactionsByPublicKeyMock.mockReturnValue([mockCoinbaseState]);
+});
+
+describe('Digital Cash Message Api', () => {
+  it('should create a correct coinbase transaction request', async () => {
+    await requestCoinbaseTransaction(
+      mockKeyPair,
+      mockKeyPair.publicKey,
+      mockTransactionValue,
+      new Hash(mockLaoId),
+    );
+
+    expect(publishMock).toBeCalledTimes(1);
+    const [channel, msgData] = publishMock.mock.calls[0];
+    expect(channel).toBe(`/root/${mockLaoId}/coin`);
+
+    const postTransaction = msgData as PostTransaction;
+    expect(postTransaction).toEqual(mockPostCoinbase);
+    checkMessageCorrectness(postTransaction);
+  });
+
+  it('should create a correct transaction request', async () => {
+    await requestSendTransaction(
+      mockPopTokenKeyPair,
+      mockKeyPair.publicKey,
+      mockTransactionValue,
+      new Hash(mockLaoId),
+    );
+
+    expect(publishMock).toBeCalledTimes(1);
+    const [channel, msgData] = publishMock.mock.calls[0];
+    expect(channel).toBe(`/root/${mockLaoId}/coin`);
+
+    const postTransaction = msgData as PostTransaction;
+    expect(postTransaction).toEqual(mockPost);
+    checkMessageCorrectness(postTransaction);
+  });
+
+  it('should throw an error when no transaction out found', async () => {
+    getTransactionsByPublicKeyMock.mockReturnValue([]);
+    try {
+      await requestSendTransaction(
+        mockPopToken,
+        mockKeyPair.publicKey,
+        mockTransactionValue,
+        new Hash(mockLaoId),
+      );
+      fail();
+    } catch (e) {
+      expect(e).toBeInstanceOf(Error);
+    }
```

Comment:
expect(...).rejects.toBeInstanceOf() is a bit shorter :)

Reply:
Seems like 
`await expect(
      await requestSendTransaction(
        mockPopToken,
        mockKeyPair.publicKey,
        -mockTransactionValue,
        new Hash(mockLaoId),
      ),
    ).rejects.toBeInstanceOf(Error);`

Does not work as expected, am I doing something wrong ?

Reply:
Should be `await expect(somePromise).rejects.toBeInstanceOf(Error);` so without the await inside the expect() :)

Reply:
Still does not catch the error :(

Reply:
Huh... What is `fail()` doing in the above code btw?

Reply:
If the request does not launch an error, the test should fail

Reply:
It is the jest fail function:
> Fails a test when called within one.

I just tried it and it makes the test fail even inside a try catch

Reply:
The problem was that the function did not return a rejecting promise but rather threw an error directly. Fixed it by making the function async :) (You could have also instead return Promise.reject(new Error()) )